api/jobset/v1alpha2/jobset_types.go

```diff
@@ -94,6 +91,9 @@ type JobSetSpec struct {
 
 	// Suspend suspends all running child Jobs when set to true.
 	Suspend *bool `json:"suspend,omitempty"`
+
+	// ManagedBy is used to indicate the controller or entity that manages a JobSet
+	ManagedBy *string `json:"managedBy,omitempty"`
```

Comment:
I would suggest that the `managedBy` value follows the format in the core k8s: https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/4368-support-managed-by-for-batch-jobs#api, with the reserved value for k8s being: `kubernetes.io/job-controller`.

The upstream validation: https://github.com/kubernetes/kubernetes/blob/f4e246bc93ffb68b33ed67c7896c379efa4207e7/pkg/apis/batch/validation/validation.go#L220-L223. I see JobSet already depends on `k8s.io/apimachinery/pkg/api/validation` so I assume `IsDomainPrefixedPath` can be used here also. 

For jobset the reserved value could be: `jobset.sigs.k8s.io/jobset-controller`, or `jobset.x-k8s.io/jobset-controller` (or something along the lines).

Reply:
This is done. I went with `jobset.sigs.k8s.io/jobset-controller` for the reserved value. I also updated the variable name in `jobset_types.go` to `JobSetControllerName` to match  the name used in the Job API.

Reply:
Sg